# List of useful bioinformatic tools for metagenomics

This is not an exhaustive list but a useful list of tools you can investigated for your analysis.
<br>
<b>Read QC and trimming</b>
- [multiQC](https://github.com/MultiQC/MultiQC)
- [fastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
- [seqkit](https://bioinf.shenwei.me/seqkit/)
- [fastp](https://github.com/OpenGene/fastp)
- [cutadapt](https://cutadapt.readthedocs.io/en/stable/)


<b>Read-based taxonomic classifiers:</b>
- [kraken2](https://github.com/DerrickWood/kraken2)
- [Centrifuge](https://github.com/DaehwanKimLab/centrifuge)
- [DeepMicrobes](https://github.com/MicrobeLab/DeepMicrobes)
- [Kaiju](https://github.com/bioinformatics-centre/kaiju)
- [singleM](https://github.com/wwood/singlem)


<b>General sequence alignment/search tools:</b>
- [diamond BLAST](https://github.com/bbuchfink/diamond)
- [minimap](https://github.com/lh3/minimap2)
- [bowtie2](https://github.com/BenLangmead/bowtie2)
- [mmseqs2](https://github.com/soedinglab/MMseqs2)


<b>Metagenome assemblers</b>

Ilummina:
- [MEGAHIT](https://github.com/voutcn/megahit)
- [metaSPAdes](https://github.com/ablab/spades)
- [velvet](https://github.com/dzerbino/velvet)

Long-read (ONT/PacBio):
- [metaFlye](https://github.com/mikolmogorov/Flye)
- [metaMBDG](https://github.com/GaetanBenoitDev/metaMDBG)


<b>Binning tools</b>
- [maxbin2](https://sourceforge.net/projects/maxbin2/)
- [metabat2](https://bitbucket.org/berkeleylab/metabat/src/master/)
- [SemiBin](https://github.com/BigDataBiology/SemiBin)
